### DISCUSSION TITLE
Mysql SET with sqlalchemy 2.x

### DISCUSSION THREAD

**Discussion by billou57** [sqlalchemy/sqlalchemy#9760] [Usage Questions]
Hello

How must we define a MySQL SET (from sqlalchemy.dialects.mysql) in sqlalchemy 2.x ? 

In my migration from version 1.4.48 to 2.0.12, I have them declared this way :

```
    whatever = mapped_column(
        SET('a', 'b', 'c', 'd'),
        nullable=False,
    )

```
They raise Mypy errors like this :

`Call to untyped function "SET" in typed context  [no-untyped-call]`

Thank you for your help

**Comment by zzzeek** [comment]
that particular SQLAlchemy source file is not currently typed.   ~~however, mypy does not normally raise errors for third party library functions.   are you running SQLAlchemy from a git checkout ?     Can you share your mypy.ini and/or pyproject.toml settings?~~

still investigating

**Comment by zzzeek** [comment]
There's discussion over this what I would consider a limitation in mypy over at https://github.com/python/mypy/issues/10757 .

SQLAlchemy's dialect SQL constructs are not fully typed right now.  So add this to your mypy.ini

```
[mypy]
strict = True
allow_untyped_calls = True
```

or use type:ignore for that specific line

**Comment by billou57** [comment]
Hello

You're right, I was setting

`disallow_untyped_calls = true`

in my pyproject.toml. I'll change that in my table definitions.
